CommonLibrary.cmake

```diff
@@ -28,6 +28,66 @@ set(COMMON_LIBRARY_TYPE SHARED CACHE STRING
   "Library type {any combination of SHARED, STATIC}")
 set_property(CACHE COMMON_LIBRARY_TYPE PROPERTY STRINGS SHARED STATIC)
 
+#-------------------------------------------------------------------------------
+# a macro which will add _D to debug library names on WIN32 builds
+# called by common_library() defined below.
+# has no effect on linux/mac unless the user sets COMMON_LIBRARY_DEBUG_SUFFIX=ON
+#-------------------------------------------------------------------------------
+MACRO (COMMON_SET_LIB_NAME_OPTIONS libtarget libname libtype)
+  # message (STATUS "${libname} libtype: ${libtype}")
+  if (${libtype} MATCHES "SHARED")
+    # default for linux/mac/unix
+    set (LIB_RELEASE_NAME "lib${libname}")
+    set (LIB_DEBUG_NAME "lib${libname}")
+    if (WIN32)
+      set (LIB_RELEASE_NAME "${libname}")
+      set (LIB_DEBUG_NAME "${libname}_D")
+    elseif (COMMON_LIBRARY_DEBUG_SUFFIX) 
+      set (LIB_RELEASE_NAME "${libname}")
+      set (LIB_DEBUG_NAME "${libname}_debug")
+    endif ()
+  else (${libtype} MATCHES "SHARED")
+    # default for linux/mac/unix
+    set (LIB_RELEASE_NAME "lib${libname}")
+    set (LIB_DEBUG_NAME "lib${libname}")
+    if (WIN32)
+      set (LIB_RELEASE_NAME "lib${libname}")
+      set (LIB_DEBUG_NAME "lib${libname}_D")
+    elseif (COMMON_LIBRARY_DEBUG_SUFFIX)
+      # if the generator supports configuration types or if the CMAKE_BUILD_TYPE has a value
+      if (CMAKE_CONFIGURATION_TYPES OR CMAKE_BUILD_TYPE)
+        set (LIB_RELEASE_NAME "${libname}")
```

Comment:
Same question here, what is the use case for this distinction with or without lib prefix on non-win32 platforms?